src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -119,18 +130,28 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String textWithImagesReplaced = userText.replaceAll(regex, replacement);
     String result = textWithImagesReplaced.replaceAll(youtube_regex, youtube_replacement);
     float sentimentScore = getSentimentScore(result);
-    boolean containsImage = !userText.equals(textWithImagesReplaced);
+    String imageURL = null
+    
+    if(blobKeys != null && !blobKeys.isEmpty()) {
```

Comment:
Before you deploy to the live server, you'll have to add another check to this if statement. Here's an example: https://github.com/KevinWorkman/GoogleCloudExamples/blob/master/blobstore/hello-world/src/main/java/io/happycoding/servlets/FormHandlerServlet.java